hr_expense_analytic_require/models/hr_expense.py

```diff
@@ -0,0 +1,12 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class HrExpense(models.Model):
+    _inherit = "hr.expense"
+
+    analytic_account_id = fields.Many2one(
+        required=True,
```

Comment:
Have this required at the View level rather than at the Model level.
This should fix the tests.